server/api/utils/singletons/k8s.py

```diff
@@ -553,24 +553,23 @@ def ensure_configmap(
         namespace = self.resolve_namespace(namespace)
         have_confmap = False
         label_name = mlrun_constants.MLRunInternalLabels.resource_name
-        full_name = f"{resource}-{name}"
+        if labels is None:
+            labels = {label_name: name}
+        else:
+            labels[label_name] = name
 
-        configmap_with_label = self.get_configmap(name, resource, namespace)
+        configmap_with_label = self.get_configmap(name, namespace)
```

Comment:
All this naming of `name` variable is a bit confusing, so what I'd suggest is having `resource_name` and `configmap_name` to reduce this confusion.